### PR TITLE
Programming questions

### PR DESCRIPTION
**Pull request by Kritika263**
Coding Questions Solved Please Review.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            <p>posted on 29 June //Category</p>
+            <p>Lorem ipsum dolor sit amet, consectetur adipisicing elit. Quibusdam vitae sit, reiciendis excepturi maiores quam rem dolorum ea nulla aut laborum placeat facere ullam odit qui est non deserunt doloribus?</p>
+            <button id="continue-btn">Continue Reading</button>
+        </div>
+    </div>
+    <div class="footer">
+       <h4>Newsfeed@2019</h4>
+    </div>
+</body>
+</html>
```
Comment (ibratnawaz) [inline index.html]: Remove this file as it was home task- 1

File: shufflearray/Arraytostring.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+function fun(arr) {
```
Comment (ibratnawaz) [inline shufflearray/Arraytostring.js]: give proper function name

File: shufflearray/Arraytostring.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+    return arr.toString();
+}
+let arr = ["h", "o", "l", "a"];
```
Comment (ibratnawaz) [inline shufflearray/Arraytostring.js]: use const instead of let and also give proper variable names

File: shufflearray/Faircalculation.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+    }
+    totalfair(x) {
+        if (x < 3) {
+            return this.dailycharges(x);
+        }
+        else if (x >= 3 && x < 7) {
+            return this.dailycharges(x) - 20
+
+        }
+        else if (x >= 7) {
```
Comment (ibratnawaz) [inline shufflearray/Faircalculation.js]: move all magic number/string in const variables.

File: shufflearray/opositeNumber.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function OppositeNumber(x){
+    x=x*(-1);
```
Comment (ibratnawaz) [inline shufflearray/opositeNumber.js]: magic number

File: shufflearray/opositeNumber.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
...
+    console.log((x));
+}
+    var x =5;
+    var a=-34.14;
```
Comment (ibratnawaz) [inline shufflearray/opositeNumber.js]: hint: 
const NUMBERS = {
   five: 5
}

File: style.css [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+    }
+    }
+    @media only screen and (max-width: 980px) {
+       .image-container{
+        width: fit-content;
+       } 
+       .content-container{
+        width: fit-content;
+       }
+    }
```
Comment (ibratnawaz) [inline style.css]: remove this file as it is previous task

File: shufflearray/Faircalculation.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+        else if (days >=numbers.Three  && days < numbers.Seven) {
+            return this.dailycharges(days) - numbers.Twenty
+
+        }
+        else if (days >= numbers.Seven) {
+            return this.dailycharges(days) - numbers.Forty;
+        }
+    }
+}
+let days;
```
Comment (ibratnawaz) [inline shufflearray/Faircalculation.js]: I think this variable is not required

File: shufflearray/Faircalculation.js [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+const fair = new Fair()
+let ans = fair.totalfair(2);
```
Comment (ibratnawaz) [inline shufflearray/Faircalculation.js]: use const

File: shufflearray/arrayshuffle.js [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+shuffleArray(inputArray);
+
+
+
+
+
+
+
+
+
```
Comment (ibratnawaz) [inline shufflearray/arrayshuffle.js]: line 11 to 18- remove these extra extra lines

File: shufflearray/opositeNumber.js [hunk 11]
```diff
@@ -0,0 +1,8 @@
+function OppositeNumber(inputnumber){
+    inputnumber=inputnumber*(-1);
+    console.log((number));
+}
+    var input1 =5;
```
Comment (ibratnawaz) [inline shufflearray/opositeNumber.js]: don't use var

File: shufflearray/opositeNumber.js [hunk 12]
```diff
@@ -0,0 +1,8 @@
...
+    var input2=-34.14;
```
Comment (ibratnawaz) [inline shufflearray/opositeNumber.js]: don't use var

**Review by ibratnawaz — COMMENTED** [review]
@Kritika263 resolve the latest review comments

File: shufflearray/Faircalculation.js (hunk 8 above)
Comment (Kritika263) [reply shufflearray/Faircalculation.js]: Not working without using this .

File: ProgrammingQuestionsAssignment2/PartitionFunction.js [hunk 13]
```diff
@@ -0,0 +1,17 @@
...
+console.log(newindex);
+function partitionOn(operation, items) {
+    let arrTrue = [];
+    let arrFalse = [];
+    for (let index = 0; index < itemsarray.length; index++) {
+        if (isEven(itemsarray[index])) arrTrue.push(itemsarray[index]);
+        else arrFalse.push(itemsarray[index]);
+    }
+    console.log(arrTrue);
+    console.log(arrFalse);
```
Comment (ibratnawaz) [inline ProgrammingQuestionsAssignment2/PartitionFunction.js]: remove unwanted console.log()

File: ProgrammingQuestionsAssignment2/PartitionFunction.js [hunk 14]
```diff
@@ -0,0 +1,17 @@
...
+    itemsarray = [...arrFalse, ...arrTrue];
+    console.log(itemsarray);
```
Comment (ibratnawaz) [inline ProgrammingQuestionsAssignment2/PartitionFunction.js]: remove unwanted console.log()

File: ProgrammingQuestionsAssignment2/WordCount.js [hunk 15]
```diff
@@ -0,0 +1,6 @@
+let inputstr = "Hi this is my java Script training";
```
Comment (ibratnawaz) [inline ProgrammingQuestionsAssignment2/WordCount.js]: use const when the variable is not changing throughout the program.

File: ProgrammingQuestionsAssignment2/PartitionFunction.js [hunk 16]
```diff
@@ -0,0 +1,17 @@
+let itemsarray = [1, 2, 3, 4, 5, 6];
+function isEven(input) { return input % 2 == 0 }
+let newindex = partitionOn(isEven, itemsarray);
```
Comment (ibratnawaz) [inline ProgrammingQuestionsAssignment2/PartitionFunction.js]: use const when as your first preference if the variable data is not changing

**Review by ibratnawaz — COMMENTED** [review]
@Kritika263  there are few small corrections, do resolve it.

**Review by ibratnawaz — COMMENTED** [review]
ALGTM

**Pull request closed without merging**